src/main/java/org/opentripplanner/routing/vertextype/StreetVertex.java

```diff
@@ -65,4 +71,18 @@ public I18NString getIntersectionName(Locale locale) {
         }
         return calculatedName;
     }
+
+    public boolean isConnectedToWalkingEdge() {
+        return this.getOutgoing().stream().anyMatch(edge ->
+            edge instanceof StreetEdge && ((StreetEdge) edge).canTraverse(WALK_TRAVERSE_MODE_SET));
```

Comment:
We could adda method, `boolean canTranverse(TraverseMode traverseMode)` in StreetEdge